src/lib/controllers/ui-event-controller.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * This is an event controller that is used currently for user interaction event, but can be adapted
+ * to handle events of any type.
+ */
+export default class UIEventController {
+  constructor () {
+    this.listeners = new Map()
+  }
+
+  /**
+   * Registers a single event listener or a group of event listeners.
+   * @param {string} name - A name of the listener or a group of listeners
+   *        that can be used to operate it (register, unregister, set, or remove).
+   * @param {string} nodeOrSelector - A CSS selector to define HTML elements that will listen to an events
+   *        or an HTML Node element.
+   * @param {Function} eventHandler - An event listener function to register.
+   * @param {PointerEvt} EventType - A constructor function of on of a custom pointer events that will do an event handling.
+   * @param {...*} eventExtraParams - Extra arguments that will be passed to the
+   *        PointerEvt constructor function after two obligatory parameters of HTML element and handler function.
+   * @return {UIEventController} An event controller instance for chaining.
+   */
+  registerListener (name, nodeOrSelector, eventHandler, EventType, ...eventExtraParams) {
+    let events
+    if (!nodeOrSelector || nodeOrSelector instanceof Node) {
+      events = [new EventType(nodeOrSelector, eventHandler, ...eventExtraParams)]
+    } else {
+      events = Array.from(document.querySelectorAll(nodeOrSelector)).map((el) => new EventType(el, eventHandler, ...eventExtraParams))
```

Comment:
see comment in the UIController about use of `document` .. I don't think we want to assume the use of the current `window.document` object, it should be explicit (see the options for the embed-lib where we allow specification of the document object.)

Reply:
I'm considering this to be a specialized DOM event controller for now. Are there any other cases we would like to use it for?

Reply:
I could be confused, but I thought the main point of the Event-based architecture was going to be as discussed at https://github.com/alpheios-project/documentation/issues/6#issuecomment-438644027  and assumed we would be registering other types of events as well via this method.

Reply:
oh wait -- was confused about the question -- I see now your previous response on the EventEmitter.  This question is about the document object. The main point here is that I don't want to force the assumption that Alpheios is bound to the main document object. To be honest, I'm not sure of the use case but since we could be embedded in a more complex application, it seemed like it would be better to make as few assumptions as possible about the embedding application.